desktop/cmp/grid/find/GridFindFieldModel.js

```diff
@@ -0,0 +1,285 @@
+        return `${match}/${count}`;
+    }
+
+    get hasResults() {
+        return !isNil(this.results) && !isEmpty(this.results);
+    }
+
+    constructor({
+        gridModel,
+        matchMode = 'startWord',
+        queryBuffer = 200,
+        includeFields,
+        excludeFields
+    }) {
+        super();
+        makeObservable(this);
+
+        this.gridModel = gridModel;
+        this.matchMode = matchMode;
+        this.queryBuffer = queryBuffer;
+        this.includeFields = includeFields;
+        this.excludeFields = excludeFields;
+
+        throwIf(!gridModel, "Must specify 'gridModel' in GridFindField.");
+
+        this.addReaction({
+            track: () => [gridModel.isReady, gridModel.store.records, gridModel.columns, gridModel.sortBy, gridModel.groupBy],
+            run: ([isReady]) => {
+                if (!isReady) return;
+                this.updateRecords();
+            }
+        });
+
+        this.addReaction({
+            track: () => [this.query, this.records],
+            run: () => this.updateResults(),
+            debounce: queryBuffer
+        });
+    }
+
+    // We allow these to be dynamically updated on every render.
+    updateProps({includeFields, excludeFields}) {
+        if (!isEqual([includeFields, excludeFields], [this.includeFields, this.excludeFields])) {
+            this.includeFields = includeFields;
+            this.excludeFields = excludeFields;
+            this.updateResults();
+        }
+    }
+
+    selectPrev() {
+        const {hasResults, results, selectedIdx, gridModel} = this;
+        if (!hasResults) return;
+        const endIdx = results.length - 1;
+        if (!isFinite(selectedIdx)) {
+            gridModel.selectAsync(results[endIdx]);
+            return;
+        }
+
+        const idx = (selectedIdx - 1) % results.length;
+        gridModel.selectAsync(results[idx < 0 ? endIdx : idx]);
+    }
+
+    selectNext() {
+        const {hasResults, results, selectedIdx, gridModel} = this;
+        if (!hasResults) return;
+        if (!isFinite(selectedIdx)) {
+            gridModel.selectAsync(results[0]);
+            return;
+        }
+
+        const idx = (selectedIdx + 1) % results.length;
+        gridModel.selectAsync(results[idx]);
+    }
+
+    //------------------------
+    // Implementation
+    //------------------------
+    @action
+    updateResults() {
+        // Track ids of matching records
+        const {query, records} = this,
+            activeFields = this.getActiveFields();
+
+        if (!query || isEmpty(activeFields)) {
+            this.results = null;
+            return;
+        }
+
+        const regex = this.getRegex(query),
+            valGetters = flatMap(activeFields, (fieldPath) => this.getValGetters(fieldPath));
+
+        this.results = records.filter(rec => {
+            return valGetters.some(fn => regex.test(fn(rec)));
+        }).map(rec => rec.id);
+    }
+
+    @action
+    updateRecords() {
```

Comment:
Pretty hardcore - this was a major subtlety (oxymoron?) that I had not considered - but of course it makes sense that would expect to walk through the records in order. I'm impressed that you worked this out w.r.t group sorting and all - nice!

I bit concerned that it could be an intensive operation for large recordsets, notably in our current client project where we were looking to use this for large grids. Might be worth profiling to ensure we are not paying more of a price than expected - could consider doing this lazily when the user first goes to actually use the control and see how that feels.